### ISSUE TITLE
Analyzer does not throw compile error for the incorrect class (whereas dart does it)

### ISSUE THREAD

**Issue by iarkh** [dart-lang/sdk#34636]
Dart SDK Version: 2.1.0-dev.6.0
OS: Windows 10

I have found this bug when verified a fix for #33477, so it's about the same code sample:
``` dart
class A<X extends B> {}
class B<X extends C> {}
class C<X extends A<B>> {}

main() {
  print("OK");
}
```
This code does not throw NoSuchMethod error now (as it's described in #33477), but still analyzer does not throw compile error whereas dart does it.

Analyzer and dart should behave in the same way and I suppose that correct behavior to break compilation here in both cases.

Now dartanalyzer silently passes here and dart throws compile error, sample output is:
> test1.dart:2:7: Error: Generic type 'B' can't be used without type arguments in the bounds of its own type variables. It is referenced indirectly through 'C'.
> Try providing type arguments to 'C' here or to some other raw types in the bounds along the reference chain.
> class B<X extends C> {}
>       ^

**Comment by stereotype441** [comment]
I believe my fix for #34635 will address this issue as well.

**Comment by stereotype441** [comment]
Fix out for review: https://dart-review.googlesource.com/c/sdk/+/78867
